### PR TITLE
cmake: fix mixing of compilers

### LINKED ISSUE #140: Mixing different compilers
In #111, when mixing gfortran with clang, clang flags get inherited, which gfortran won't understand.

### PR DESCRIPTION
**Pull request by junghans**
Fix #140 (hopefully)

### CODE REVIEW

File: cmake/FindKOKKOS.cmake [hunk 1]
```diff
@@ -38,6 +38,7 @@ if(KOKKOS_SETTINGS_DIR AND KOKKOS_INCLUDE_DIR AND KOKKOS_LIBRARY)
     IMPORTED_LOCATION ${KOKKOS_LIBRARY}
     INTERFACE_INCLUDE_DIRECTORIES ${KOKKOS_INCLUDE_DIR}
     INTERFACE_COMPILE_OPTIONS "$<$<COMPILE_LANGUAGE:CXX>:${KOKKOS_CXX_FLAGS_WITHOUT_INCLUDES}>"
+    IMPORTED_LINK_INTERFACE_LANGUAGES "CXX"
```
Comment (dalg24) [inline cmake/FindKOKKOS.cmake]: Did you mean to quote `CXX`?

Comment (junghans) [reply cmake/FindKOKKOS.cmake]: I think it doesn't matter.

**Comment by junghans** [comment]
> Does it fix the issue or not?

I tested `clang++` together with `gfortran` and it worked.

**Comment by junghans** [comment]
@guangyechen can you try this with PGI?

**Comment by guangyechen** [comment]
It does not seem to work with clang+pgfortran.
[error_clang_pgfortran.txt](https://github.com/ECP-copa/Cabana/files/3307939/error_clang_pgfortran.txt)

**Comment by junghans** [comment]
Did you compile kokkos with clang or pgi?

**Comment by guangyechen** [comment]
Clang.

**Comment by junghans** [comment]
@guangyechen we could just put exclude mixing PGI with anything ;-)

**Comment by dalg24** [comment]
I suspect it is mixing libstdc++ that PGI is taking from the GCC 4.8.5 on the system

**Pull request merged**